packages/office-ui-fabric-react/src/components/Nav/Nav.tsx

```diff
@@ -303,9 +303,15 @@ export class Nav extends BaseComponent<INavProps, INavState> implements INav {
   }
 
   private _onLinkExpandClicked(link: INavLink, ev: React.MouseEvent<HTMLElement>): void {
+    const { onLinkExpandClick } = this.props;
+
     link.isExpanded = !link.isExpanded;
     this.setState({ isLinkExpandStateChanged: true });
 
+    if (onLinkExpandClick) {
```

Comment:
1 more suggestion. If we put this callback above the setState, we can then wrap the setState in a if (!ev.defaultPrevented) so that the callback has the ability to prevent the default action if necessary.

Reply:
Like?
```typescript
    if (onLinkExpandClick) {
      onLinkExpandClick(ev, link);
    } else {
      link.isExpanded = !link.isExpanded;
      this.setState({ isLinkExpandStateChanged: true });
    }
```

Reply:
Yup :) Thanks